### ISSUE TITLE
Cannot open Trash

### ISSUE THREAD

**Issue by smithtim** [mate-desktop/caja#1161]
#### Expected behaviour

(a) Double-clicking the Trash icon on the desktop opens the Trash.

(b) From a Caja window, clicking Trash in the side pane opens the Trash.

#### Actual behaviour

(a) An error pops up: `Could not display "trash:".  The file is of an unknown type`

(b) An error pops up: `The folder contents could not be displayed.  Sorry, could not display all the contents of "trash": Operation not supported`

#### Steps to reproduce

Double-click the Trash icon on the desktop or click Trash in the side pane of a Caja window.

#### MATE general version

MATE 1.20.4

#### Package version

Caja 1.20.3

#### Linux Distribution

NixOS

**Comment by lukefromdc** [comment]
By any chance are you running Caja as root?  For some reason neither Caja nor Nemo nor Nautilus can open the trash when running as root, showing exactly the error you describe

**Comment by smithtim** [comment]
Yes, actually I am!  So that's the issue.  Do you know why it doesn't work as root?  If it affects other file managers too, perhaps a shared component is responsible.  Can we identify that component and report the bug somewhere?

**Comment by lukefromdc** [comment]
So far I have no idea. Running` sudo caja` from terminal and trying to open the trash does not generate any warnings in terminal.

**Comment by selectiveduplicate** [comment]
I can reproduce the second behaviour and get the second error message when running Caja as ``root``.

**Comment by rbuj** [comment]
It's a glib issue:
```
$ LANG=C sudo gio list trash://
gio: trash://: Operation not supported
```

**Comment by smithtim** [comment]
Update: I can now open the Trash as root!  I added the following to my `configuration.nix` (NixOS configuration file), trying to fix another issue:

```
services.gnome3.gvfs.enable = true;
environment.variables.GIO_EXTRA_MODULES = [
  "${pkgs.gnome3.gvfs}/lib/gio/modules"
];
```

After rebuilding and rebooting, it worked.  So maybe gio just wasn't set up or configured correctly?
